### PR TITLE
Implement a TestClient to remove all the stubbing

### PR DESCRIPTION
**Pull request by ChrisBr**
This PR does basically two things:

## Implement `expect_metric` matchers
All the stubbing of `write_point` was an eyesore for me so this is an attempt to implement a `TestClient` class and some custom matchers for it.

Instead of doing ` expect_any_instance_of(InfluxDB::Client).to receive(:write_point).with("rails", data)` we can do now something more natural like `expect_metric(tags: {}, values: {})`.

Ideally this would live in `influxdb-ruby` at some point.

## Migrate unit specs to request specs

The unit specs were quite difficult to read and maintain. I migrated them to request specs which will give us more flexibility to refactor in the future. The request spesc are now also very clean and can be used for documentation.

```ruby
  it "writes metric" do
    get "/metrics"

    expect_metric(
      name:   "rails",
      tags:   a_hash_including(
        method:          "MetricsController#index",
        hook:            "process_action",
        status:          200,
        format:          :html,
        http_method:     "GET",
        additional_tag:  :value,
        server:          Socket.gethostname,
        app_name:        :app_name,
        tags_middleware: :tags_middleware
      ),
      values: a_hash_including(
        additional_value: :value,
        request_id:       :request_id,
        view:             be_between(1, 20),
        db:               be_between(1, 20),
        controller:       be_between(1, 20)
      )
    )
  end
```

### CODE REVIEW

**Comment by hennevogel** [comment]
I like it :+1:

**Comment by ChrisBr** [comment]
There is one spec failing for the SQL metrics on Rails 5.0 and 5.1, the difference is that the `class_name` and `name` is `SQL` instead of `Metric`. Need to dig into Rails what is going on. If there is a difference, what do you reckon @hennevogel 

* just dropping checking these keys
* do a conditional in our specs, something like

```ruby
class_name = {
  "5.0" => "SQL",
  "6.0" => "Metric"
}

it "records metrics" do
  get "/metrics"

  expect_metric(
    tags: a_hash_including(
        class_name: class_name[Rails.version]
     )
  )
end
```

File: spec/requests/render_collection_metrics_spec.rb [hunk 1]
```diff
@@ -0,0 +1,67 @@
+require File.dirname(__FILE__) + "/../spec_helper"
+
+RSpec.describe "Render collection metrics", type: :request do
```
Comment (hennevogel) [inline spec/requests/render_collection_metrics_spec.rb]: Can we stick to subscriber names somehow so it's clear what we test?

Comment (hennevogel) [reply spec/requests/render_collection_metrics_spec.rb]: Like `RSpec.describe "RenderSubscriber collection"`....

Comment (hennevogel) [reply spec/requests/render_collection_metrics_spec.rb]: Actually `RSpec.describe "Action View collection metrics"` makes more sense I guess

File: spec/requests/render_partial_metrics_spec.rb [hunk 2]
```diff
@@ -0,0 +1,63 @@
+require File.dirname(__FILE__) + "/../spec_helper"
+
+RSpec.describe "Render partial metrics", type: :request do
```
Comment (hennevogel) [inline spec/requests/render_partial_metrics_spec.rb]: RSpec.describe "Action View partial metrics"

File: spec/requests/render_template_metrics_spec.rb [hunk 3]
```diff
@@ -0,0 +1,63 @@
+require File.dirname(__FILE__) + "/../spec_helper"
+
+RSpec.describe "Render template metrics", type: :request do
```
Comment (hennevogel) [inline spec/requests/render_template_metrics_spec.rb]: RSpec.describe "Action View template metrics"

File: spec/requests/request_metrics_spec.rb [hunk 4]
```diff
@@ -0,0 +1,83 @@
+require File.dirname(__FILE__) + "/../spec_helper"
+
+RSpec.describe "Request metrics", type: :request do
```
Comment (hennevogel) [inline spec/requests/request_metrics_spec.rb]: RSpec.describe "Action Controller metrics"

File: spec/requests/sql_metrics_spec.rb [hunk 5]
```diff
@@ -0,0 +1,104 @@
+require File.dirname(__FILE__) + "/../spec_helper"
+
+RSpec.describe "SQL metrics", type: :request do
```
Comment (hennevogel) [inline spec/requests/sql_metrics_spec.rb]: `RSpec.describe "Active Record metrics"`

**Review by hennevogel — APPROVED** [review]

**Pull request merged**